frame/contracts-registry/src/tests.rs

```diff
@@ -335,6 +341,49 @@ fn creator_can_remove_contract() {
 	});
 }
 
+#[test]
+fn anyone_can_remove_contract() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::signed(2);
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			origin.clone(),
+			vec![].try_into().unwrap(),
+			vec![].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 97u64);
+		assert!(Contracts::<Test>::get(0).is_some());
+		assert_ok!(ContractsRegistry::remove_contract(RuntimeOrigin::signed(3), 0));
+		assert!(Contracts::<Test>::get(0).is_none());
+		assert_eq!(Balances::free_balance(2), 100u64);
+	});
+}
+
+#[test]
+fn creator_can_not_remove_contract() {
+	new_test_ext().execute_with(|| {
+		let origin = RuntimeOrigin::signed(2);
+
+		let parties = account("a", 0, SEED);
+		let parties_origin = RuntimeOrigin::signed(parties);
+
+
+		assert_eq!(Balances::free_balance(2), 100u64);
+		assert_ok!(ContractsRegistry::create_contract(
+			origin.clone(),
+			vec![].try_into().unwrap(),
+			vec![parties].try_into().unwrap()
+		));
+		assert_eq!(Balances::free_balance(2), 97u64);
+		assert!(Contracts::<Test>::get(0).is_some());
+		assert_ok!(ContractsRegistry::party_sign_contract(parties_origin.into(), 0));
+		assert_noop!(ContractsRegistry::remove_contract(origin.into(), 0), Error::<Test>::ContractInUse);
```

Comment:
Missing test for case when there are no parties signatures, but there is a judge signature.